tools/convertUsageStatsLogFile.php

```diff
@@ -0,0 +1,570 @@
+
+                $this->setAssoc($assocType, $op, $args, $newEntry);
+                if (!array_key_exists('assocType', $newEntry)) {
+                    echo "The URL {$entryData['url']} in the line number {$lineNumber} was not considered.\n";
+                    continue;
+                }
+            } else {
+                continue;
+            }
+
+            // Geo data
+            $country = $region = $city = null;
+            if ($ipNotHashed === 1) {
+                $statisticsHelper = new StatisticsHelper();
+                $site = Application::get()->getRequest()->getSite();
+                [$country, $region, $city] = $statisticsHelper->getGeoData($site, $context, $ip, $hashedIp, false);
+            }
+            $newEntry['country'] = $country;
+            $newEntry['region'] = $region;
+            $newEntry['city'] = $city;
+
+            // institutions IDs
+            $institutionIds = [];
+            if ($ipNotHashed === 1 && $context->isInstitutionStatsEnabled($site)) {
+                $institutionIds = $statisticsHelper->getInstitutionIds($context->getId(), $ip, $hashedIp, false);
+            }
+            $newEntry['institutionIds'] = $institutionIds;
+
+            $newEntry['version'] = Registry::get('appVersion');
+
+            // write to a new file
+            $newLogEntry = json_encode($newEntry) . PHP_EOL;
+            file_put_contents($newFilePath, $newLogEntry, FILE_APPEND);
+            $isSuccessful = true;
+        }
+        fclose($fhandle);
+
+        if ($isSuccessful) {
+            $renameFilePath = $archivePath . '/' . $path_parts['filename'] . '_old.log';
+            if (!rename($filePath, $renameFilePath)) {
+                echo "Cound not rename the file {$filePath} to {$renameFilePath}.\n";
+                exit(1);
+            } else {
+                echo "Old file was renamed to {$renameFilePath}.\n";
+            }
+            if (!rename($newFilePath, $filePath)) {
+                echo "Cound not rename the new file {$newFilePath} to {$filePath}.\n";
+                exit(1);
+            } else {
+                echo "File {$filePath} is converted.\n";
+            }
+            if ($extension == 'gz') {
+                try {
+                    $renameFilePath = $fileMgr->gzCompressFile($renameFilePath);
+                    $filePath = $fileMgr->gzCompressFile($filePath);
+                } catch (Exception $e) {
+                    printf($e->getMessage() . "\n");
+                    exit(1);
+                }
+            }
+        } else {
+            echo "File {$filePath} could not be successfully converted.\n";
+            exit(1);
+        }
+    }
+
+    /**
+     * Get data from the passed log entry.
+     */
+    private function getDataFromLogEntry(string $entry): array
+    {
+        $entryData = [];
+        if (preg_match(self::PARSEREGEX, $entry, $m)) {
+            $associative = count(array_filter(array_keys($m), 'is_string')) > 0;
```

Comment:
I don't suppose it does any harm, but is the idea here that users might customized `self::PARSEREGEX` to use numeric indexes?

Reply:
I am not sure about numeric indexes :thinking: but yes, the idea here is that users might customized the regex -- above all if their apache log files are in a different format that this here.
I also do not think it can cause any harm -- it is only applied to that file that they want to convert...